src/lnclient/LNClient.ts

```diff
@@ -0,0 +1,94 @@
+import { Invoice, LightningAddress } from "@getalby/lightning-tools";
+import { Nip47MakeInvoiceRequest, Nip47PayInvoiceRequest } from "../nwc/types";
+import { NewNWCClientOptions, NWCClient } from "../nwc/NWCClient";
+import { ReceiveInvoice } from "./ReceiveInvoice";
+import { Amount, resolveAmount } from "./Amount";
+
+export type LNClientCredentials = string | NWCClient | NewNWCClientOptions;
+
+/**
+ * A simple lightning network client to interact with your lightning wallet
+ */
+export class LNClient {
+  readonly nwcClient: NWCClient;
+
+  /**
+   * Create a new LNClient
+   * @param credentials credentials to connect to a NWC-based wallet. This can be a NWC connection string e.g. nostr+walletconnect://... or an existing NWC Client. Learn more at https://nwc.dev
+   */
+  constructor(credentials: LNClientCredentials) {
+    if (typeof credentials === "string") {
+      this.nwcClient = new NWCClient({
+        nostrWalletConnectUrl: credentials,
+      });
+    } else if (credentials instanceof NWCClient) {
+      this.nwcClient = credentials;
+    } else {
+      this.nwcClient = new NWCClient(credentials);
+    }
+  }
+
+  /**
+   * Make a payment
+   * @param recipient a BOLT-11 invoice or lightning address
+   * @param amount the amount to pay, only required if paying to a lightning address or the amount is not specified in the BOLT 11 invoice.
+   * @param args additional options, e.g. to store metadata on the payment
+   * @returns the receipt of the payment, and details of the paid invoice.
+   */
+  async pay(
+    recipient: string,
+    amount?: Amount,
+    args?: Omit<Nip47PayInvoiceRequest, "invoice" | "amount">,
```

Comment:
```suggestion
    recipient: string,
    amount?: Amount,
    comment?: string,
    args?: Omit<Nip47PayInvoiceRequest, "invoice" | "amount">,
```

do you think it makes sense to get the comment in there which we pass on when requesting the invoice from the LN address? 
it's often used to add a message.

Reply:
I'm not sure about putting it as its own argument since it only applies to payments to a LN address.

To be consistent with how we handle metadata for received payments in the hub, I think setting args as `{metadata: {comment, payer_data}}` would do it, I just need to update the typings here so it's more obvious for users.

Related PR: https://github.com/getAlby/go/pull/247

Reply:
https://github.com/getAlby/js-sdk/pull/356